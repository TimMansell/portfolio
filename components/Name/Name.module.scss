@import "styles/scssvars";
@import "styles/utilities/pxtorem";

.name {
  composes: hero shadowlg from "styles/helpers/text.scss";
  padding: var(--vr-md);
  font-size: pxtorem(64);
  letter-spacing: pxtorem(5);
  text-transform: uppercase;
  animation: title 1s ease-in-out;
  animation-delay: var(--animation-name-delay);
  animation-fill-mode: forwards;

  @media (min-width: $bp-md) {
    font-size: pxtorem(96);
  }
}

@keyframes title {
  from {
    opacity: 0;
    transform: scale(0.95);
  }

  to {
    opacity: 1;
    transform: scale(1);
  }
}
