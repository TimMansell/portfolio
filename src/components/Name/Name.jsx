import styles from './Name.module.scss';

export const Name = () => {
  return <h1 className={styles.name}>I&apos;m Tim Mansell</h1>;
};

export default Name;
