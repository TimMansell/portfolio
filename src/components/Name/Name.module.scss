@import 'variables';
@import 'utilities/pxtorem';
@import 'utilities/text-shadow';

.name {
  color: var(--secondary-text-colour);
  padding: var(--vr-md);
  font-size: pxtorem(64);
  letter-spacing: pxtorem(5);
  line-height: 1;
  text-transform: uppercase;
  text-shadow: text-shadow(1, 4, var(--colour-1-alt-2));
  opacity: 0;
  animation: title 1s ease-in-out;
  animation-delay: var(--animation-name-delay);
  animation-fill-mode: forwards;

  @media (min-width: $bp-md) {
    text-shadow: text-shadow(1, 4, var(--tertiary-text-colour));
    font-size: pxtorem(96);
  }
}

@keyframes title {
  from {
    opacity: 0;
    transform: scale(0.95);
  }

  to {
    opacity: 1;
    transform: scale(1);
  }
}
