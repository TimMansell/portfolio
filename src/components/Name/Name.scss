@import 'variables';
@import 'utilities/breakpoints';

.name {
	color: $secondary-text-colour;
	margin: 15px 0;
	padding: 0 25px;
	font-size: 45px;
	font-weight: 900;
	letter-spacing: 5px;
	line-height: 45px;
	text-transform: uppercase;
	text-shadow: 3px 3px darken($primary-colour, 10%);
	opacity: 0;

	animation: title 2s ease-in-out;
	animation-delay: 3.65s;
	animation-fill-mode: forwards;

	@include media-breakpoint-up(md){
		text-shadow: 2px 2px $tertiary-text-colour;
		font-size: 55px;
		line-height: 55px;
	}
}

@keyframes title {
	from {
		opacity: 0;
		transform: translateY(15px);
	}

	to {
		opacity: 1;
		transform: translateY(0);
	}
}