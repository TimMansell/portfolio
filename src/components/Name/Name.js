import React from 'react';

import './Name.scss';

export const Name = () => {
  return <h1 className="name">I&apos;m Tim Mansell</h1>;
};

export default Name;
