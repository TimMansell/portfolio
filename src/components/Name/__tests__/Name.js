import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Name from '../Name';

describe('Name', () => {
  it('should render my component', () => {
    const wrapper = shallow(<Name />);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<Name/>).toJSON();

    expect(snapshot).toMatchSnapshot();
  });
});
