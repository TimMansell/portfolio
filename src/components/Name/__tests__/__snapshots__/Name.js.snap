// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name should match snapshot 1`] = `
<h1
  className="name"
>
  I'm Tim Mansell
</h1>
`;
